plugins/module_utils/entrust_cagw/api.py

```
@@ -0,0 +1,347 @@
+                self.request.client_key = cagw_api_cert_key
+        else:
+            raise SessionConfigurationException(to_native("Client certificate for authentication to the API must be provided."))
+
+        # set up the spec
+        cagw_api_specification_path = self.get_config("cagw_api_specification_path")
+
+        if not cagw_api_specification_path.startswith("http") and not os.path.isfile(cagw_api_specification_path):
+            raise SessionConfigurationException(to_native("OpenAPI specification was not found at location {0}.".format(cagw_api_specification_path)))
+        if not valid_file_format.match(cagw_api_specification_path):
+            raise SessionConfigurationException(to_native("OpenAPI specification filename must end in .json, .yml or .yaml"))
+
+        self.verify = True
+
+        if cagw_api_specification_path.startswith("http"):
+            try:
+                http_response = Request().open(method="GET", url=cagw_api_specification_path)
+                http_response_contents = http_response.read()
+                if cagw_api_specification_path.endswith(".json"):
+                    self._spec = json.load(http_response_contents)
+                elif cagw_api_specification_path.endswith(".yml") or cagw_api_specification_path.endswith(".yaml"):
+                    self._spec = yaml.safe_load(http_response_contents)
+            except HTTPError as e:
+                raise SessionConfigurationException(to_native("Error downloading specification from address '{0}', received error code '{1}'".format(
+                    cagw_api_specification_path, e.getcode())))
+        else:
+            with open(cagw_api_specification_path) as f:
+                if ".json" in cagw_api_specification_path:
+                    self._spec = json.load(f)
+                elif ".yml" in cagw_api_specification_path or ".yaml" in cagw_api_specification_path:
+                    self._spec = yaml.safe_load(f)
```

If none of the two branches matches, you should react as well (like raise an exception "unknown file extension" or something?).